third-culture-software/bhima
# Automatically merge duplicate lots

This PR adds a function to automatically merge duplicate lots (where unambiguous).   All lots that have the same lot label, inventory_uuid, and expiration date will be merged by this function.

Closed #5370

This function is in the main menu of the 'Stock > Find Duplicate Lots' page.

**WARNING:** This is a first cut.  Do not release it yet!
- It still needs some work on the autoMerge() function in server/controllers/stock/lots.js
- The function works but does not refresh the duplicate lots list.  I think this has to do with the way the individual merge() functions are being used.

**TESTING**
- Use one of the production sites (eg, IMCK)
- Be sure to enable the permissions for the test user to access the "Find Duplicate Lots" page (under the Stock menu).
- In the Stock > Find duplicate lots, note the number of duplicate lots found (at the bottom of the page)
- Try the "Automatically Merge Lots" in the main [Menu] button
- It should work, but you will to have to refresh the page to see the updated total.  With the IMCK db, this went from 423 down to 201 after executing this function.
